src/sentry/tasks/store.py

```diff
@@ -421,7 +422,12 @@ def save_event(cache_key=None, data=None, start_time=None, event_id=None,
     finally:
         if cache_key:
             default_cache.delete(cache_key)
-            attachment_cache.delete(cache_key)
+
+            # For the unlikely case that we did not manage to persist the
```

Comment:
Personally, I don’t think this edge case is worth handling. Everything is TTLd, so worst case, it hangs around for a bit before expiring.

Reply:
It needs to be handled either way because `event` can be unbound. So it's just a question of 

```python
event is None or features.has('organizations:event-attachments', event.project.organization, actor=None)
```

or

```python
event is not None and features.has('organizations:event-attachments', event.project.organization, actor=None)
```